src/lib/interfaces/mail-log.ts

```diff
@@ -0,0 +1,19 @@
+import { App } from './app';
+import { User } from './user';
+
+export interface MailLog {
+  email_id: string;
+  app: App;
+  user: User;
+  sender: string;
+  recipient: string;
+  reply_to: string;
+  create_date: string;
+  open_date: string;
+  status: string;
+  status_localized: string;
+  reject_reason: string;
```

Comment:
Similarly I found that `open_date` and `reject_reason` can be null, when an email hasn't been opened, or rejected respectively.